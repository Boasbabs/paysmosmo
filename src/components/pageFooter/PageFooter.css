.page-footer {
    color: #989898;
}

.page-footer--top{
    background-color: #2D2E37;
    min-height: 200px;
    color: #989898;
    padding-top: 3.5em;
    padding-bottom: 3.5em;
}

.page-footer--bottom {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #26262E;
    text-transform: uppercase;
    min-height: 40px;
    text-align: center;
    padding-top: 2.5em;
    padding-bottom: 2.5em;
}
.footer--title {
    color: #989898;
    font-size: 20px;
    font-weight: bold;
    letter-spacing: 1px;
    margin-bottom: 1.1em;
    padding-left: 0.8em;
    text-transform: uppercase;
}
.footer--subtitle {
    color: #989898;
    font-size: 16px;
    font-weight: bold;
    letter-spacing: 1px;
    margin-bottom: 1.1em;
}
.footer--link a {
    color: #989898;
    padding-left: 0 !important;
}
.footer--icon svg {
    height: 1.5em;
    width: 1.5em;
    padding: 0;
    margin: auto 0;
}

.footer--paragraph {
    color: #989898;
    font-size: 14px;
    line-height: 1.2;
    padding-left: 1.1em;
}