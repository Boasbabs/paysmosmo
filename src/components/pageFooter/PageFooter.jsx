import React, { Component } from "react";
import { Layout, Icon, Row, Col, List } from "antd";
import "./PageFooter.css";

const { Footer } = Layout;
const helpMenuData = ["Privacy Policy", "How it works", "Terms & Conditions"];
const contactMenuData = ["Registration", "About Us", "Contact Us"];

class PageFooter extends Component {
  state = {};
  render() {
    return (
      <Layout className="page-footer">
        <Footer className="page-footer--top">
          <Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}>
            <Col xs={24} sm={24} md={12} lg={7} xl={8}>
              <h3 className="footer--title">Paysmosmo</h3>

              <p className="footer--paragraph">
                Help give out inventory loans to <br />
                cooperative members with ease.
              </p>
            </Col>
            <Col xs={24} sm={24} md={12} lg={7} xl={4}>
              <h4 className="footer--subtitle">Contact</h4>
              <List
                size="small"
                split={false}
                bordered={false}
                dataSource={contactMenuData}
                renderItem={item => (
                  <List.Item className="footer--link">
                    <a href="/" target="_blank" rel="noopener noreferrer">
                      {item}
                    </a>
                  </List.Item>
                )}
              />
            </Col>
            <Col xs={24} sm={24} md={12} lg={7} xl={4}>
              <h4 className="footer--subtitle">Help</h4>
              <List
                size="small"
                split={false}
                bordered={false}
                dataSource={helpMenuData}
                renderItem={item => (
                  <List.Item className="footer--link">
                    <a href="/" target="_blank" rel="noopener noreferrer">
                      {item}
                    </a>
                  </List.Item>
                )}
              />
            </Col>
            <Col xs={24} sm={24} md={12} lg={7} xl={4}>
              <h4 className="footer--subtitle">Stay Connected</h4>
              <Row>
                <Col xs={8} sm={8} md={6} lg={6} xl={6}>
                  <List.Item className="footer--link ">
                    <a href="/" target="_blank" rel="noopener noreferrer">
                      <Icon type="facebook" className="footer--icon" />
                    </a>
                  </List.Item>
                </Col>
                <Col xs={8} sm={8} md={6} lg={6} xl={6}>
                  <List.Item className="footer--link ">
                    <a href="/" target="_blank" rel="noopener noreferrer">
                      <Icon type="twitter" className="footer--icon" />
                    </a>
                  </List.Item>
                </Col>
                <Col xs={8} sm={8} md={6} lg={6} xl={6}>
                  <List.Item className="footer--link ">
                    <a href="/" target="_blank" rel="noopener noreferrer">
                      <Icon
                        type="instagram"
                        theme="filled"
                        className="footer--icon"
                      />
                    </a>
                  </List.Item>
                </Col>
              </Row>
            </Col>
            <Col xs={24} sm={24} md={12} lg={7} xl={4}>
              <h4 className="footer--subtitle">Back to top ^</h4>
            </Col>
          </Row>
        </Footer>
        <Layout className="page-footer--bottom">&copy; 2018 paysmosmo</Layout>
      </Layout>
    );
  }
}

export default PageFooter;
