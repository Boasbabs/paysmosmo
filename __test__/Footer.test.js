import React from "react";
import { create } from "react-test-renderer";
import PageFooter from "../src/components/pageFooter";

describe("PageFooter component", () => {
  test("matches its snapshot", () => {
    const component = create(<PageFooter />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
