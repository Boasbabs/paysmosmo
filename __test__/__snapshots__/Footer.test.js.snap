// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter component matches its snapshot 1`] = `
<section
  className="page-footer ant-layout"
>
  <footer
    className="page-footer--top ant-layout-footer"
  >
    <div
      className="ant-row"
    >
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-8"
      >
        <h3
          className="footer--title"
        >
          Paysmosmo
        </h3>
        <p
          className="footer--paragraph"
        >
          Help give out inventory loans to 
          <br />
          cooperative members with ease.
        </p>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-4"
      >
        <h4
          className="footer--subtitle"
        >
          Contact
        </h4>
        <div
          className="ant-list ant-list-sm"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Registration
                </a>
              </div>
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  About Us
                </a>
              </div>
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Contact Us
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-4"
      >
        <h4
          className="footer--subtitle"
        >
          Help
        </h4>
        <div
          className="ant-list ant-list-sm"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div
              className="ant-spin-container"
            >
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Privacy Policy
                </a>
              </div>
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  How it works
                </a>
              </div>
              <div
                className="ant-list-item footer--link"
              >
                <a
                  href="/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Terms & Conditions
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-4"
      >
        <h4
          className="footer--subtitle"
        >
          Stay Connected
        </h4>
        <div
          className="ant-row"
        >
          <div
            className="ant-col-xs-8 ant-col-sm-8 ant-col-md-6 ant-col-lg-6 ant-col-xl-6"
          >
            <div
              className="ant-list-item footer--link "
            >
              <a
                href="/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  aria-label="icon: facebook"
                  className="anticon anticon-facebook footer--icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="facebook"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-32 736H663.9V602.2h104l15.6-120.7H663.9v-77.1c0-35 9.7-58.8 59.8-58.8h63.9v-108c-11.1-1.5-49-4.8-93.2-4.8-92.2 0-155.3 56.3-155.3 159.6v89H434.9v120.7h104.3V848H176V176h672v672z"
                    />
                  </svg>
                </i>
              </a>
            </div>
          </div>
          <div
            className="ant-col-xs-8 ant-col-sm-8 ant-col-md-6 ant-col-lg-6 ant-col-xl-6"
          >
            <div
              className="ant-list-item footer--link "
            >
              <a
                href="/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  aria-label="icon: twitter"
                  className="anticon anticon-twitter footer--icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="twitter"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0 0 75-94 336.64 336.64 0 0 1-108.2 41.2A170.1 170.1 0 0 0 672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 0 0-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 0 1-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 0 1-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
                    />
                  </svg>
                </i>
              </a>
            </div>
          </div>
          <div
            className="ant-col-xs-8 ant-col-sm-8 ant-col-md-6 ant-col-lg-6 ant-col-xl-6"
          >
            <div
              className="ant-list-item footer--link "
            >
              <a
                href="/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  aria-label="icon: instagram"
                  className="anticon anticon-instagram footer--icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="instagram"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 0 1-47.9 47.9z"
                    />
                  </svg>
                </i>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-4"
      >
        <h4
          className="footer--subtitle"
        >
          Back to top ^
        </h4>
      </div>
    </div>
  </footer>
  <section
    className="page-footer--bottom ant-layout"
  >
    Â© 2018 paysmosmo
  </section>
</section>
`;
